### ISSUE TITLE
Proposal: Invocation image OCI runtime requirements extension

### ISSUE THREAD

**Issue by jlegrone** [cnabio/cnab-spec#363]
For OCI-based invocation images, there are cases in which invocation image authors may need to express certain runtime requirements such as memory quota or the ability to mount a docker socket.

A well-known custom extension for defining these requirements would ideally

- Support scoping requirements to an invocation image, not the bundle (multiple invocation images can be present in the same bundle, and may have different requirements)
  - Do [invocation images](https://github.com/cnabio/cnab-spec/blob/master/101-bundle-json.md#invocation-images) need an embedded `custom` map?
- Allow for future extensibility as new use cases are defined
- Be generic enough to be used by multiple bundle builders, CLIs, and execution runtimes

**Comment by technosophos** [comment]
I agree -- I think this is going to be increasingly important for not just cloud cases, but for constrained providers (edge, IoT) as well.
